secboot/secboot_hooks.go

```diff
@@ -98,17 +97,22 @@ func SealKeysWithFDESetupHook(runHook fde.RunSetupHookFunc, keys []SealKeyReques
 		if primaryKey == nil {
 			primaryKey = primaryKeyOut
 		}
-		if err := skr.BootstrappedContainer.AddKey(skr.SlotName, unlockKey); err != nil {
-			return err
-		}
-		writer := sb.NewFileKeyDataWriter(skr.KeyFile)
-		if err := protectedKey.WriteAtomic(writer); err != nil {
-			return err
-		}
-	}
-	if primaryKey != nil && params.AuxKeyFile != "" {
-		if err := osutil.AtomicWriteFile(params.AuxKeyFile, primaryKey, 0600, 0); err != nil {
-			return fmt.Errorf("cannot write the policy auth key file: %v", err)
+		if skr.KeyFile != "" {
+			if err := skr.BootstrappedContainer.AddKey(skr.SlotName, unlockKey); err != nil {
+				return err
+			}
+			writer := sb.NewFileKeyDataWriter(skr.KeyFile)
+			if err := protectedKey.WriteAtomic(writer); err != nil {
+				return err
+			}
+		} else {
+			tokenWriter, err := skr.BootstrappedContainer.AddKeyAndGetTokenWriter(skr.SlotName, unlockKey)
```

Comment:
now that I see how it's used, would it make sense to not have `AddKeyAndGetTokenWriter` on the container, but rather a more generic `AddKey()` which depending on the container type would either return a token writer or `FileKeyDataWriter` ?

Reply:
Well, I was asked not to do that.

Reply:
We discussed what to do here. We decided to split `AddKeyAndGetTokenWriter`.